Improve usability

hickeyma opened the pull request:
Had good discussion with @timroster about usability. Here are the usability issues being covered: 

- Provide inline swagger comment for post api
- Reference the swagger pet store example for user guidance
- Point the user that the visitor object should be in the base openapi
file
- Python code is now 2.7 & 3 compatible (PR #5 )
- Update the base openapi to be standard YAML
- Remove deprecated verification section from challenge 1
- Add note to clear server cache before testing

Closes #2 
Closes #1

hickeyma committed:
Improve usability

- Provide inline swagger comment for post api
- Reference the swagger pet store example for user guidance
- Point the user that the visitor object should be in the base openapi
file
- Document that Python code is 2.7 compatibile only (will make 3
compatibile in future PR)
- Update the base openapi to be standard YAML
- Remove derecated verification section from challenge 1
- Add note to clear server cache before testing

timroster commented on docs/draftOpenAPIChallenge.md:
```diff
@@ -23,15 +23,18 @@ npm install -g swagger-inline
 
-The comments should be added to the sample app `server.js` file where the app API endpoints are defined. The comments should be added above the:
-- [visitors POST API](https://github.ibm.com/developer-first-guild/get-started-openapi/blob/master/server.js#L38)
+The comments should be added to the sample app `server.js` file where the app API endpoints are defined. The [visitors POST API](https://github.ibm.com/developer-first-guild/get-started-openapi/blob/master/server.js#L38) comment has already being added for guidance.
```
With the addition of the example for the /post - the line numbers increment by 22. New numbers are 60, 88, 104 and 134 for the respective endpoints.

timroster replied:
Rephrase last sentence to "...comment has already been added for guidance."

timroster replied:
I started to add the updates from this file to Influitive, but until server.js is merged this won't work so holding off on that until the PR is merged.

hickeyma replied:
Good comments. I updated:
- The `server.js` line numbers
- Changed 'being' to 'been'

Good idea to hold back on updating the coder program, until the PR gets merged.

timroster commented on docs/generateClientCodeChallenge.md:
````diff
@@ -70,6 +70,8 @@ With the application still running, you can run the [test code](https://github.i
 python ./test/test_apis.py
 ```
 
+*Note: Requires Python 2.7 and the app should be restarted before running tests to clear the cache.*
````
Updated on Influitive, however, I've moved this up into the other dependency note about urllib3, certifi and six as it's best dealt with before running any python code:

"...may include `urllib3`, `certifi` and `six`. The tools provided in the repository are intended for use with Python 2.7."

hickeyma replied:
Good idea. Updated.

henrynash reviewed (approved):
LGTM - some nice improvements here.

hickeyma committed:
Update after review

Update for following review comments:
- https://github.com/IBM/get-started-openapi/pull/3#pullrequestreview-207127667
- https://github.com/IBM/get-started-openapi/pull/3#pullrequestreview-207128219

hickeyma commented:
@timroster Thanks for review and comments. Updated and ready for review again.

hickeyma committed:
Remove Python 2.7 references

Compatability added for Python 2.7 and 3 in PR #5

hickeyma commented:
Updates after rebase. Ready for review.

henrynash commented:
Still LGTM

timroster commented:
LGTM

timroster commented:
Updated IBM Coder (with corrected links)

hickeyma commented:
Thanks @henrynash for merge.
